torch/_inductor/comm_analysis.py

```diff
@@ -216,8 +217,170 @@ def estimate_nccl_collective_runtime_nccl_estimator(snode) -> Optional[float]:
     return est_time_ms
 
 
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+
+class ConnectionType(Enum):
+    CURRENT_DEVICE = "current_device"
+    NVLINK = "nvlink"
+    INFINIBAND = "infiniband"
+
+
+class Connection(ABC):
+    """Base class for GPU interconnect connections"""
+
+    @property
+    @abstractmethod
+    def type(self) -> ConnectionType:
+        """Connection type"""
+
+    @property
+    @abstractmethod
+    def bandwidth(self) -> float:
+        """Bidirectional bandwidth in GB/s"""
+
+
+@dataclass
+class CurrentDevice(Connection):
+    """Dummy Implementation"""
+
+    type: ConnectionType = ConnectionType.CURRENT_DEVICE
+    bandwidth: float = 0.0
+
+
+@dataclass
+class NVLinkConnection(Connection):
+    """NVLink connection for intra-node GPU communication"""
+
+    version: str
+    num_links: int
+    # Bidirectional bandwidth per link (GB/s)
+    BANDWIDTH_PER_LINK = {
+        "1.0": 40.0,  # 20 GB/s unidirectional × 2
+        "2.0": 50.0,  # 25 GB/s unidirectional × 2
+        "3.0": 50.0,  # 25 GB/s unidirectional × 2 (A100)
+        "4.0": 50.0,  # 25 GB/s unidirectional × 2 (H100, H200)
+        "5.0": 400,  # 200 GB/s unidirectional x 2 (GB200)
+    }
+
+    def __post_init__(self) -> None:
+        if self.version not in self.BANDWIDTH_PER_LINK:
+            raise ValueError(
+                f"Unknown NVLink version: {self.version}. "
+                f"Supported versions: {list(self.BANDWIDTH_PER_LINK.keys())}"
+            )
+        # Calculate total bidirectional bandwidth
+        self._bandwidth = self.BANDWIDTH_PER_LINK[self.version] * self.num_links
+
+    @property
+    def bandwidth(self) -> float:
+        """Bidirectional bandwidth in GB/s"""
+        return self._bandwidth
+
+    @property
+    def type(self) -> ConnectionType:
+        return ConnectionType.NVLINK
+
+    def __str__(self) -> str:
+        return f"NVLink v{self.version} ({self.num_links} links, {self.bandwidth:.0f} GB/s bidirectional)"
+
+    def __repr__(self) -> str:
+        return f"NVLinkConnection(version='{self.version}', num_links={self.num_links}, bandwidth={self.bandwidth:.0f})"
+
+
+@dataclass
+class InfiniBandConnection(Connection):
+    """InfiniBand connection for inter-node communication"""
+
+    rate: float = 200.0  # Default (4xHDR), ibstat Rate output, Gbps per direction
+    num_ports: int = 4
+
+    def __post_init__(self) -> None:
+        self._bandwidth = self.rate / 8 * self.num_ports * 2
```

Comment:
Nit: have it be listed along with the other instance variables like rate and num_ports but set Type dataclass.InitVar[float]. That way static type checking will work and it knows the attribute exists, but it will never be returned with .fields or be set by the ctor.